File: app/src/main/java/com/demo/justapp/exchanger/di/application/HasComponent.java
package com.demo.justapp.exchanger.di.application;

/**
 * Класс, реализующий этот интерфейс, возвращает Dagger компонент
 *
 * @param <T> класс компонента
 * @author Sergey Rodionov
 */
public interface HasComponent<T> {

    /**
     * @return возвращает запрашиваемый компонент
     */
    T getComponent();
}

File: app/src/main/kotlin/com/demo/justapp/exchanger/di/application/HasComponent.kt
package com.demo.justapp.exchanger.di.application

/**
 * Класс, реализующий этот интерфейс, возвращает Dagger компонент
 *
 * @param T класс компонента
 */
interface HasComponent<T> {

    fun getComponent(): T
}